### PR TITLE
Backend 2

### PR DESCRIPTION
**Pull request by Couzy21**
Made changes to request syntax to update folder and ensured process is efficient for multiple documents and template additions

### CODE REVIEW

File: app/utils/mongo_db_connection.py [hunk 1]
```diff
@@ -1104,3 +1121,17 @@ def reminder_func(reminder):
         )
     ]
     return reminder_list
+
+
+def process_folders_to_item(ids, folder_id, add_item_to_folder):
+    processes = []
+
+    for id in ids:
+        process = multiprocessing.Process(
```
Comment (WawinyEdwin) [inline app/utils/mongo_db_connection.py:1130]: What is this multiprocessing all about?

File: app/views.py [hunk 2]
```diff
@@ -1268,34 +1273,27 @@ def folder_update(request, folder_id):
         return Response(settings, status.HTTP_200_OK)
```
Comment (WawinyEdwin) [inline app/views.py:1273]: What is setting definition doing in a folder update endpoint?. 
Are you just copy pasting this blindly?

File: app/views.py [hunk 3]
```diff
@@ -1309,4 +1307,6 @@ def all_folders(request, company_id):
             cache.set(cache_key, folders_list, timeout=60)
         except:
             return Response(status.HTTP_500_INTERNAL_SERVER_ERROR)
+    if request.method == "PUT":
```
Comment (WawinyEdwin) [inline app/views.py:1310]: What is this method and code block doing here? Remove it

File: app/views.py [hunk 4]
```diff
@@ -1268,34 +1273,27 @@ def folder_update(request, folder_id):
...
             return Response("Folder Data is Required", status.HTTP_400_BAD_REQUEST)
+        items = form["items"]
         old_folder = get_folder_object(folder_id)
         old_folder["folder_name"] = form["folder_name"]
-        old_folder["data"].append(
-            {"item_id:": form["item_id"], "item_type": form["item_type"]}
-        )
+        old_folder["data"].extend(items)
         updt_folder = json.loads(update_folder(folder_id, old_folder))
-        if request.data["item_type"] == "document":
```
Comment (WawinyEdwin) [inline app/views.py:1280]: All these nested if statements not required, this logic should be takes outside this view function.
A view should only accept response and return the necessary response.. Keep business logic out of your views?

File: app/utils/mongo_db_connection.py [hunk 5]
```diff
@@ -977,6 +985,15 @@ def add_document_to_folder(document_id, folder):
 
 
 def add_template_to_folder(template_id, folder):
+    old_template = get_template_object(template_id)
```
Comment (WawinyEdwin) [inline app/utils/mongo_db_connection.py:988]: Nested if statements is bad practice

File: app/utils/mongo_db_connection.py [hunk 6]
```diff
@@ -960,6 +959,15 @@ def delete_document(document_id, data_type):
 
 
 def add_document_to_folder(document_id, folder):
+    old_document = get_document_object(document_id)
```
Comment (WawinyEdwin) [inline app/utils/mongo_db_connection.py:962]: Again, nested if statements is bad practice

File: app/tests.py [hunk 7]
```diff
@@ -85,4 +91,22 @@ def test_custom_time_limit_outside_range(self):
 
 
 if __name__ == "__main__":
-    unittest.main()
\ No newline at end of file
+    unittest.main()
+"""
+
+
+class ExampleModelTestCase(TestCase):
```
Comment (WawinyEdwin) [inline app/tests.py:98]: This Test class does nothing here.

What is it exactly that you are testing??

**Review by WawinyEdwin — CHANGES_REQUESTED** [review]

File: app/views.py (hunk 3 above)
Comment (Couzy21) [reply app/views.py:1310]: Okay

File: app/views.py (hunk 4 above)
Comment (Couzy21) [reply app/views.py:1280]: The business logics here are small, and I didn't much find it necessary to create much external imports them, those i found excess were kept from it.

**Review by WawinyEdwin — APPROVED** [review]
We can work with this

**Pull request merged**